[CI] PartialHitCountCollectorTests testCollectedHitCountEarlyTerminated failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/dgoewdilmzalk/tests/:server:test/org.elasticsearch.search.query.PartialHitCountCollectorTests/testCollectedHitCountEarlyTerminated

**Reproduction line:**
```
./gradlew ':server:test' --tests "org.elasticsearch.search.query.PartialHitCountCollectorTests.testCollectedHitCountEarlyTerminated" -Dtests.seed=8F487EFE6D51AA33 -Dtests.locale=es-CR -Dtests.timezone=Europe/Moscow -Druntime.java=20
```

**Applicable branches:**
main, 8.9

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.search.query.PartialHitCountCollectorTests&tests.test=testCollectedHitCountEarlyTerminated

**Failure excerpt:**
```
java.lang.AssertionError: expected:<1> but was:<3>

  at __randomizedtesting.SeedInfo.seed([8F487EFE6D51AA33:FF8FBDB932638B7F]:0)
  at org.junit.Assert.fail(Assert.java:88)
  at org.junit.Assert.failNotEquals(Assert.java:834)
  at org.junit.Assert.assertEquals(Assert.java:645)
  at org.junit.Assert.assertEquals(Assert.java:631)
  at org.elasticsearch.search.query.PartialHitCountCollectorTests.testCollectedHitCountEarlyTerminated(PartialHitCountCollectorTests.java:112)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:578)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1623)

```

--- a/server/src/test/java/org/elasticsearch/search/query/NonCountingTermQuery.java
+++ b/server/src/test/java/org/elasticsearch/search/query/NonCountingTermQuery.java
@@ -1,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.query;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.FilterWeight;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.ScoreMode;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.search.Weight;
+
+import java.io.IOException;
+
+/**
+ * A {@link TermQuery} which does not retrieve hit count from Weight#count for reproducible tests.
+ * Using this query we will never early-terminate the collection phase because we can already
+ * get the document count from the term statistics of each segment.
+ */
+class NonCountingTermQuery extends TermQuery {
+
+    NonCountingTermQuery(Term term) {
+        super(term);
+    }
+
+    public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float boost) throws IOException {
+        Weight w = super.createWeight(searcher, scoreMode, boost);
+        return new FilterWeight(w) {
+            public int count(LeafReaderContext context) throws IOException {
+                return -1;
+            }
+        };
+    }
+}

--- a/server/src/test/java/org/elasticsearch/search/query/PartialHitCountCollectorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/query/PartialHitCountCollectorTests.java
@@ -12,18 +12,22 @@ import org.apache.lucene.document.Document;
 import org.apache.lucene.document.Field;
 import org.apache.lucene.document.StringField;
 import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.LeafReaderContext;
 import org.apache.lucene.index.Term;
 import org.apache.lucene.search.BooleanClause;
 import org.apache.lucene.search.BooleanQuery;
 import org.apache.lucene.search.IndexSearcher;
 import org.apache.lucene.search.MatchAllDocsQuery;
 import org.apache.lucene.search.Query;
+import org.apache.lucene.search.ScoreMode;
 import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.search.Weight;
 import org.apache.lucene.store.Directory;
 import org.apache.lucene.tests.index.RandomIndexWriter;
 import org.elasticsearch.test.ESTestCase;
 
 import java.io.IOException;
+import java.util.List;
 
 public class PartialHitCountCollectorTests extends ESTestCase {
 
@@ -109,7 +113,25 @@ public class PartialHitCountCollectorTests extends ESTestCase {
         int totalHitsThreshold = randomInt(2);
         PartialHitCountCollector partialHitCountCollector = new PartialHitCountCollector(totalHitsThreshold);
         searcher.search(query, partialHitCountCollector);
-        assertEquals(totalHitsThreshold, partialHitCountCollector.getTotalHits());
-        assertTrue(partialHitCountCollector.hasEarlyTerminated());
+
+        try {
+            assertEquals(totalHitsThreshold, partialHitCountCollector.getTotalHits());
+            assertTrue(partialHitCountCollector.hasEarlyTerminated());
+        } catch (AssertionError e) {
+            // if we get very unlucky with the document distribution, we can get more than one segment with only one document
+            // in this case early termination might not kick in because we never collect
+            // check if this was the case here and disregard error in that case
+            Weight weight = searcher.createWeight(query, ScoreMode.TOP_DOCS, 1.0f);
+            List<LeafReaderContext> leafContexts = searcher.getLeafContexts();
+            int totalHitsWithoutCollecting = 0;
+            for (LeafReaderContext leafContext : leafContexts) {
+                int count = weight.count(leafContext);
+                if (count != -1) {
+                    totalHitsWithoutCollecting += count;
+                }
+            }
+            assertTrue(totalHitsWithoutCollecting > totalHitsThreshold);
+            assertFalse(partialHitCountCollector.hasEarlyTerminated());
+        }
     }
 }

--- a/server/src/test/java/org/elasticsearch/search/query/PartialHitCountCollectorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/query/PartialHitCountCollectorTests.java
@@ -16,6 +16,7 @@ import org.apache.lucene.index.LeafReaderContext;
 import org.apache.lucene.index.Term;
 import org.apache.lucene.search.BooleanClause;
 import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.FilterWeight;
 import org.apache.lucene.search.IndexSearcher;
 import org.apache.lucene.search.MatchAllDocsQuery;
 import org.apache.lucene.search.Query;
@@ -27,7 +28,6 @@ import org.apache.lucene.tests.index.RandomIndexWriter;
 import org.elasticsearch.test.ESTestCase;
 
 import java.io.IOException;
-import java.util.List;
 
 public class PartialHitCountCollectorTests extends ESTestCase {
 
@@ -105,33 +105,33 @@ public class PartialHitCountCollectorTests extends ESTestCase {
     }
 
     public void testCollectedHitCountEarlyTerminated() throws Exception {
-        Query query = new BooleanQuery.Builder().add(new TermQuery(new Term("string", "a1")), BooleanClause.Occur.SHOULD)
-            .add(new TermQuery(new Term("string", "b3")), BooleanClause.Occur.SHOULD)
-            .add(new TermQuery(new Term("string", "b6")), BooleanClause.Occur.SHOULD)
+        Query query = new BooleanQuery.Builder().add(termQuery(new Term("string", "a1")), BooleanClause.Occur.SHOULD)
+            .add(termQuery(new Term("string", "b3")), BooleanClause.Occur.SHOULD)
+            .add(termQuery(new Term("string", "b6")), BooleanClause.Occur.SHOULD)
             .build();
         // there's three docs matching the query: any totalHitsThreshold lower than 3 will trigger early termination
         int totalHitsThreshold = randomInt(2);
         PartialHitCountCollector partialHitCountCollector = new PartialHitCountCollector(totalHitsThreshold);
         searcher.search(query, partialHitCountCollector);
 
-        try {
-            assertEquals(totalHitsThreshold, partialHitCountCollector.getTotalHits());
-            assertTrue(partialHitCountCollector.hasEarlyTerminated());
-        } catch (AssertionError e) {
-            // if we get very unlucky with the document distribution, we can get more than one segment with only one document
-            // in this case early termination might not kick in because we never collect
-            // check if this was the case here and disregard error in that case
-            Weight weight = searcher.createWeight(query, ScoreMode.TOP_DOCS, 1.0f);
-            List<LeafReaderContext> leafContexts = searcher.getLeafContexts();
-            int totalHitsWithoutCollecting = 0;
-            for (LeafReaderContext leafContext : leafContexts) {
-                int count = weight.count(leafContext);
-                if (count != -1) {
-                    totalHitsWithoutCollecting += count;
-                }
+        assertEquals(totalHitsThreshold, partialHitCountCollector.getTotalHits());
+        assertTrue(partialHitCountCollector.hasEarlyTerminated());
+    }
+
+    /**
+     * Create a {@link TermQuery} which cannot terminate collection early
+     */
+    private TermQuery termQuery(Term term) {
+        return new TermQuery(term) {
+            public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float boost) throws IOException {
+
+                Weight w = super.createWeight(searcher, scoreMode, boost);
+                return new FilterWeight(w) {
+                    public int count(LeafReaderContext context) throws IOException {
+                        return -1;
+                    }
+                };
             }
-            assertTrue(totalHitsWithoutCollecting > totalHitsThreshold);
-            assertFalse(partialHitCountCollector.hasEarlyTerminated());
-        }
+        };
     }
 }

--- a/server/src/test/java/org/elasticsearch/search/query/PartialHitCountCollectorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/query/PartialHitCountCollectorTests.java
@@ -44,7 +44,9 @@ public class PartialHitCountCollectorTests extends ESTestCase {
         for (int i = 0; i < numDocs; i++) {
             Document doc = new Document();
             doc.add(new StringField("string", "a" + i, Field.Store.NO));
-            doc.add(new StringField("string", "b" + i, Field.Store.NO));
+            if (i < 3) {
+                doc.add(new StringField("string", "foo", Field.Store.NO));
+            }
             writer.addDocument(doc);
         }
         if (randomBoolean()) {
@@ -95,7 +97,7 @@ public class PartialHitCountCollectorTests extends ESTestCase {
 
     public void testCollectedHitCount() throws Exception {
         Query query = new BooleanQuery.Builder().add(new TermQuery(new Term("string", "a1")), BooleanClause.Occur.SHOULD)
-            .add(new TermQuery(new Term("string", "b3")), BooleanClause.Occur.SHOULD)
+            .add(new TermQuery(new Term("string", "a3")), BooleanClause.Occur.SHOULD)
             .build();
         // there's two docs matching the query: any totalHitsThreshold greater than or equal to 2 will non cause early termination
         PartialHitCountCollector partialHitCountCollector = new PartialHitCountCollector(randomIntBetween(2, Integer.MAX_VALUE));
@@ -105,10 +107,7 @@ public class PartialHitCountCollectorTests extends ESTestCase {
     }
 
     public void testCollectedHitCountEarlyTerminated() throws Exception {
-        Query query = new BooleanQuery.Builder().add(termQuery(new Term("string", "a1")), BooleanClause.Occur.SHOULD)
-            .add(termQuery(new Term("string", "b3")), BooleanClause.Occur.SHOULD)
-            .add(termQuery(new Term("string", "b6")), BooleanClause.Occur.SHOULD)
-            .build();
+        Query query = new NonCountingTermQuery(new Term("string", "foo"));
         // there's three docs matching the query: any totalHitsThreshold lower than 3 will trigger early termination
         int totalHitsThreshold = randomInt(2);
         PartialHitCountCollector partialHitCountCollector = new PartialHitCountCollector(totalHitsThreshold);
@@ -119,19 +118,22 @@ public class PartialHitCountCollectorTests extends ESTestCase {
     }
 
     /**
-     * Create a {@link TermQuery} which cannot terminate collection early
+     * A {@link TermQuery} which cannot terminate collection early
      */
-    private TermQuery termQuery(Term term) {
-        return new TermQuery(term) {
-            public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float boost) throws IOException {
+    public static class NonCountingTermQuery extends TermQuery {
 
-                Weight w = super.createWeight(searcher, scoreMode, boost);
-                return new FilterWeight(w) {
-                    public int count(LeafReaderContext context) throws IOException {
-                        return -1;
-                    }
-                };
-            }
-        };
+        public NonCountingTermQuery(Term term) {
+            super(term);
+        }
+
+        public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float boost) throws IOException {
+            Weight w = super.createWeight(searcher, scoreMode, boost);
+            return new FilterWeight(w) {
+                public int count(LeafReaderContext context) throws IOException {
+                    return -1;
+                }
+            };
+        }
     }
+
 }

--- a/server/src/test/java/org/elasticsearch/search/query/PartialHitCountCollectorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/query/PartialHitCountCollectorTests.java
@@ -118,11 +118,13 @@ public class PartialHitCountCollectorTests extends ESTestCase {
     }
 
     /**
-     * A {@link TermQuery} which cannot terminate collection early
+     * A {@link TermQuery} which does not retrieve hit count from Weight#count for reproducible tests.
+     * Using this query we will never early-terminate the collection phase because we can already
+     * get the document count from the term statistics of each segment.
      */
-    public static class NonCountingTermQuery extends TermQuery {
+    static class NonCountingTermQuery extends TermQuery {
 
-        public NonCountingTermQuery(Term term) {
+        NonCountingTermQuery(Term term) {
             super(term);
         }
 

--- a/server/src/test/java/org/elasticsearch/search/query/PartialHitCountCollectorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/query/PartialHitCountCollectorTests.java
@@ -12,17 +12,13 @@ import org.apache.lucene.document.Document;
 import org.apache.lucene.document.Field;
 import org.apache.lucene.document.StringField;
 import org.apache.lucene.index.IndexReader;
-import org.apache.lucene.index.LeafReaderContext;
 import org.apache.lucene.index.Term;
 import org.apache.lucene.search.BooleanClause;
 import org.apache.lucene.search.BooleanQuery;
-import org.apache.lucene.search.FilterWeight;
 import org.apache.lucene.search.IndexSearcher;
 import org.apache.lucene.search.MatchAllDocsQuery;
 import org.apache.lucene.search.Query;
-import org.apache.lucene.search.ScoreMode;
 import org.apache.lucene.search.TermQuery;
-import org.apache.lucene.search.Weight;
 import org.apache.lucene.store.Directory;
 import org.apache.lucene.tests.index.RandomIndexWriter;
 import org.elasticsearch.test.ESTestCase;
@@ -115,27 +111,6 @@ public class PartialHitCountCollectorTests extends ESTestCase {
 
         assertEquals(totalHitsThreshold, partialHitCountCollector.getTotalHits());
         assertTrue(partialHitCountCollector.hasEarlyTerminated());
-    }
-
-    /**
-     * A {@link TermQuery} which does not retrieve hit count from Weight#count for reproducible tests.
-     * Using this query we will never early-terminate the collection phase because we can already
-     * get the document count from the term statistics of each segment.
-     */
-    static class NonCountingTermQuery extends TermQuery {
-
-        NonCountingTermQuery(Term term) {
-            super(term);
-        }
-
-        public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float boost) throws IOException {
-            Weight w = super.createWeight(searcher, scoreMode, boost);
-            return new FilterWeight(w) {
-                public int count(LeafReaderContext context) throws IOException {
-                    return -1;
-                }
-            };
-        }
     }
 
 }

--- a/server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java
+++ b/server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java
@@ -77,6 +77,7 @@ import org.elasticsearch.search.internal.ReaderContext;
 import org.elasticsearch.search.internal.ScrollContext;
 import org.elasticsearch.search.internal.SearchContext;
 import org.elasticsearch.search.internal.ShardSearchContextId;
+import org.elasticsearch.search.query.PartialHitCountCollectorTests.NonCountingTermQuery;
 import org.elasticsearch.search.rank.RankShardContext;
 import org.elasticsearch.search.rank.RankShardResult;
 import org.elasticsearch.search.sort.SortAndFormats;
@@ -434,8 +435,8 @@ public class QueryPhaseTests extends IndexShardTestCase {
      */
     public void testTerminateAfterSize0NoHitCountShortcut() throws Exception {
         indexDocs();
-        BooleanQuery bq = new BooleanQuery.Builder().add(new TermQuery(new Term("foo", "bar")), Occur.SHOULD)
-            .add(new TermQuery(new Term("foo", "baz")), Occur.SHOULD)
+        BooleanQuery bq = new BooleanQuery.Builder().add(new NonCountingTermQuery(new Term("foo", "bar")), Occur.SHOULD)
+            .add(new NonCountingTermQuery(new Term("foo", "baz")), Occur.SHOULD)
             .build();
         {
             TestSearchContext context = createContext(newContextSearcher(reader), bq);

--- a/server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java
+++ b/server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java
@@ -435,11 +435,9 @@ public class QueryPhaseTests extends IndexShardTestCase {
      */
     public void testTerminateAfterSize0NoHitCountShortcut() throws Exception {
         indexDocs();
-        BooleanQuery bq = new BooleanQuery.Builder().add(new NonCountingTermQuery(new Term("foo", "bar")), Occur.SHOULD)
-            .add(new NonCountingTermQuery(new Term("foo", "baz")), Occur.SHOULD)
-            .build();
+        Query query = new NonCountingTermQuery(new Term("foo", "bar"));
         {
-            TestSearchContext context = createContext(newContextSearcher(reader), bq);
+            TestSearchContext context = createContext(newContextSearcher(reader), query);
             context.terminateAfter(1);
             context.setSize(0);
             QueryPhase.executeQuery(context);
@@ -450,7 +448,7 @@ public class QueryPhaseTests extends IndexShardTestCase {
         }
         // test interaction between trackTotalHits and terminateAfter
         {
-            TestSearchContext context = createContext(newContextSearcher(reader), bq);
+            TestSearchContext context = createContext(newContextSearcher(reader), query);
             context.terminateAfter(10);
             context.setSize(0);
             context.trackTotalHitsUpTo(-1);
@@ -461,7 +459,7 @@ public class QueryPhaseTests extends IndexShardTestCase {
             assertThat(context.queryResult().topDocs().topDocs.scoreDocs.length, equalTo(0));
         }
         {
-            TestSearchContext context = createContext(newContextSearcher(reader), bq);
+            TestSearchContext context = createContext(newContextSearcher(reader), query);
             context.terminateAfter(10);
             context.setSize(0);
             // track total hits is lower than terminate_after
@@ -474,7 +472,7 @@ public class QueryPhaseTests extends IndexShardTestCase {
             assertThat(context.queryResult().topDocs().topDocs.scoreDocs.length, equalTo(0));
         }
         {
-            TestSearchContext context = createContext(newContextSearcher(reader), bq);
+            TestSearchContext context = createContext(newContextSearcher(reader), query);
             context.terminateAfter(10);
             context.setSize(0);
             // track total hits is higher than terminate_after

--- a/server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java
+++ b/server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java
@@ -77,7 +77,6 @@ import org.elasticsearch.search.internal.ReaderContext;
 import org.elasticsearch.search.internal.ScrollContext;
 import org.elasticsearch.search.internal.SearchContext;
 import org.elasticsearch.search.internal.ShardSearchContextId;
-import org.elasticsearch.search.query.PartialHitCountCollectorTests.NonCountingTermQuery;
 import org.elasticsearch.search.rank.RankShardContext;
 import org.elasticsearch.search.rank.RankShardResult;
 import org.elasticsearch.search.sort.SortAndFormats;

